src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -162,14 +167,24 @@ private IDataEncoder GetDataEncoder(TransferMode requestedTransferMode, Transfer
 
         public async Task StopAsync() => await StopAsyncCore().ForceAsync();
 
-        private Task StopAsyncCore() => _connection.StopAsync();
+        private async Task StopAsyncCore()
+        {
+            await _connection.StopAsync();
+
+            if (_readingTask != null)
```

Comment:
I think you have to capture `_readingTask` before calling `_connection.StopAsync` otherwise someone could start a new connection in the closed callback and set `_readingTask` for the new connection

Reply:
Yea, this part of the code is racy. I took a look last night and since the receive loop moved out of the place that had all of the locks and synchronization it's a bit messy.